builtin/providers/aws/resource_aws_route53_zone_test.go

```diff
@@ -206,6 +206,42 @@ func TestAccAWSRoute53Zone_private_region(t *testing.T) {
 	})
 }
 
+func TestAccAWSRoute53Zone_privateToPublic(t *testing.T) {
+	var before, after route53.GetHostedZoneOutput
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_route53_zone.main",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckRoute53ZoneDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRoute53PrivateZoneConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRoute53ZoneExists("aws_route53_zone.main", &before),
+					testAccCheckRoute53ZoneAssociatesWithVpc("aws_vpc.main", &before),
+				),
+			},
+			{
+				Config: testAccRoute53PrivateZoneConfigToPublic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRoute53ZoneExists("aws_route53_zone.main", &after),
+					testAccCheckRoute53ZoneRecreated(t, &before, &after),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckRoute53ZoneRecreated(t *testing.T, before, after *route53.GetHostedZoneOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if before.HostedZone.Id == after.HostedZone.Id {
```

Comment:
We're checking pointers here and not values, so I don't believe this is confirming what we want it to